### DISCUSSION TITLE
LFN: too many dirs open

### DISCUSSION THREAD

**Discussion by j2969719** [dosemu2/comcom64#91] [Q&A]
When I try to compile freecom in dosemu (`build -r russian`) using both comcom32/64 as a shell for some reason I get this error:
`ERROR: LFN: too many dirs open`
It's strange that when using freecom, the compilation completes successfully.

**Comment by j2969719** [comment]
tnx, the error disappeared and compilation continued, but at the end wlink started asking me for .lib after `wlink /ma/nologo @command.rsp`

**Comment by stsp** [comment]
Andrew, maybe you can add the
freecom build to CI?

**Reply by andrewbird** [reply]
Okay will do something later.

**Reply by stsp** [reply]
Maybe you can check what is the
problem there with building it with
comcom64? I never built it.

**Reply by andrewbird** [reply]
okay.

**Comment by j2969719** [comment]
btw, it seems that the problems with freecom compilation almost gone, the only thing that is interrupted is the lines in the batch file containing `copy` with `/b` switch

**Reply by stsp** [reply]
OK, added the code to ignore /b.

**Reply by j2969719** [reply]
comcom doesnt like this line

`copy /b shell\command.exe + infores + criter\criter1 + criter\criter %CMD_NAME%`

Too many parameters ...

**Reply by stsp** [reply]
Please open a comcom ticket for this.

**Comment by j2969719** [comment]
I tried to compile with the current version of comcom: there also seems to be another issue with lines
```
type tools.m1 >tools.mak
..\utils\mktools.exe >>tools.mak
type tools.m2 >>tools.mak
```
it seems `type` adds extra carriage return? character. instead `0d 0a` tools.mak contains `0d 0d 0a` sequences

**Reply by stsp** [reply]
Should now be fixed.

**Reply by j2969719** [reply]
Thanks, now freecom compiled successfully with comcom64+fdpp!

**Reply by stsp** [reply]
Good, so I wonder why Andrew got
stuck with that, while you already succeeded. :)

**Reply by andrewbird** [reply]
I raised an issue #97 for why Freecom build does not work for my test under FDPP/Comcom.
